gmusicapi/clients/mobileclient.py

```diff
@@ -43,6 +43,14 @@ def _ensure_device_id(self, device_id=None):
 
         return device_id
 
+    def _validate_device_id(self, device_id):
+        """Ensure that a given device_id belongs to the user supplying it."""
+        if device_id in [d['id'][2:] if d['id'].startswith('0x') else d['id']
+                         for d in self.get_registered_devices()]:
+            return device_id
+        else:
+            raise ValueError('Invalid device_id.')
```

Comment:
Good point around putting it in the output.

Maybe we make a new exception, then put the device ids in the message and also in a field? That way if the client code doesn't handle the exception the user will see the list of ids to pick from, but the client code could always handle it and do something else (prompt the user with them, automatically pick one, etc).

I'm imagining something along the lines of
```python
class InvalidDeviceId(Exception):
    def __init__(self, message, valid_device_ids):
        super(ValidationError, self).__init__(message)
        self.valid_device_ids = valid_device_ids
```

I can't remember if that's exactly right for custom exceptions (I normally just inherit and don't bother overriding anything). There may actually be an example in gmusicapi already?